peregrine/iqgen/bits/doppler_const.py

```diff
@@ -0,0 +1,196 @@
+      phase and delay.
+    speed_mps : float
+      Speed of satellite in meters per second.
+    '''
+    super(Doppler, self).__init__()
+    self.distance0_m = distance0_m
+    self.speed_mps = speed_mps
+    self.c1 = self.distance0_m / (scipy.constants.c + self.speed_mps)
+    self.c2 = self.speed_mps / (scipy.constants.c + self.speed_mps)
+
+  def computeDistanceM(self, svTime_s):
+    '''
+    Computes distance to satellite in meters.
+
+    Parameters
+    ----------
+    svTime_s : float
+      Time in seconds at which distance is computed. Please note that  is not
+      a time of the observer.
+
+    Returns
+    -------
+    float
+      Distance to satellite in meters.
+    '''
+    return self.distance0_m + self.speed_mps * svTime_s
+
+  def computeSpeedMps(self, svTime_s):
+    '''
+    Computes speed of satellite in meters per second.
+
+    Parameters
+    ----------
+    svTime_s : float
+      Time in seconds at which speed is computed. Please note that  is not
+      a time of the observer.
+
+    Returns
+    -------
+    float
+      Speed of satellite in meters per second.
+    '''
+    return self.speed_mps
+
+  def computeSvTimeS(self, userTime_s):
+    '''
+    Computes SV time from a user time.
+
+    The computation is a solution of the formulae:
+    T_user = T_sv + D(T_sv)/c, for D(t) = D_0 + V*T_sv
+
+    Parameters
+    ----------
+    userTime_s : float
+      Observer's time in seconds
+
+    Returns
+    -------
+    float
+      Satellite vehicle's time at which signal has been generated.
+    '''
+
+    tau_s = self.c1 + self.c2 * userTime_s
+    svTime_s = userTime_s - tau_s
+
+    return svTime_s
+
+  def computeBatch(self,
+                   userTime0_s,
+                   n_samples,
+                   amplitude,
+                   planFrequency_hz,
+                   ifFrequency_hz,
+                   message,
+                   code):
+    '''
+    Computes signal samples for the doppler object.
+
+    Parameters
+    ----------
+    userTime0_s : float
+      Observer's time in seconds of the beginning of the interval.
+    n_samples : int
+      Number of samples to generate
+    amplitude : float
+      Signal amplitude.
+    planFrequency_hz : float
+      Central carrier frequency in hertz
+    ifFrequency_hz: float
+      Intermediate frequency in hertz
+    message : object
+      Message object for providing access to symbols
+    code : object
+      PRN code object for providing access to chips
+
+    Returns
+    -------
+    signal : numpy.ndarray(n_samples, dtype=float)
+      Generated samples
+    userTimeX_s : float
+      End of interval time in seconds
+    chipAll_idx : numpy.ndarray(n_samples, dtype=float)
+      Code chip phases for the samples
+    chips : numpy.ndarray(n_samples, dtype=int)
+      Code combined with data
+    '''
+    deltaUserTime_s = float(n_samples) / if_iface.Chip.SAMPLE_RATE_HZ
```

Comment:
It is a bad design. Not sure how to handle it correctly: the Chip type contains sample generation parameters.